src/SerializeAST.cc

```diff
@@ -589,31 +622,107 @@ sos::Object WrapResource(const Resource& resource)
     return resourceObject;
 }
 
-sos::Object WrapResourceGroup(const ResourceGroup& resourceGroup)
+sos::Object WrapResourceGroup(const Element& resourceGroup)
 {
     sos::Object resourceGroupObject;
 
     // Name
-    resourceGroupObject.set(SerializeKey::Name, sos::String(resourceGroup.name));
+    resourceGroupObject.set(SerializeKey::Name, sos::String(resourceGroup.attributes.name));
 
-    // Description
-    resourceGroupObject.set(SerializeKey::Description, sos::String(resourceGroup.description));
-
-    // Resources
+    // Description && Resources
+    std::string description;
     sos::Array resources;
 
-    for (Resources::const_iterator it = resourceGroup.resources.begin();
-         it != resourceGroup.resources.end();
+    for (Elements::const_iterator it = resourceGroup.content.elements().begin();
+         it != resourceGroup.content.elements().end();
          ++it) {
 
-        resources.push(WrapResource(*it));
+        if (it->element == Element::ResourceElement) {
+            resources.push(WrapResource(it->content.resource));
+        }
+        else if (it->element == Element::CopyElement) {
+
+            if (!description.empty()) {
+                TwoNewLines(description);
+            }
+
+            description += it->content.copy;
+        }
     }
 
+    resourceGroupObject.set(SerializeKey::Description, sos::String(description));
     resourceGroupObject.set(SerializeKey::Resources, resources);
 
     return resourceGroupObject;
 }
 
+sos::Object WrapDataStructureContent(const DataStructure& dataStructure)
+{
+    sos::Object dataStructureObject;
+
+    // Source
+    dataStructureObject.set(SerializeKey::Source, WrapNamedType(dataStructure.source));
+
+    // Resolved
+    dataStructureObject.set(SerializeKey::Resolved, sos::Object());
+
+    return dataStructureObject;
+}
+
+sos::Object WrapElement(const Element& element)
+{
+    // Resource Element is special case
+    if (element.element == Element::ResourceElement) {
+        return WrapResource(element.content.resource, true);
```

Comment:
OK, so in a new "DOM"  it should be true to serialize the `element` property, in old resource group it should be `false` since it is not needed there. Correct? 

This should be explained in the comment – that this parameter is for wrapping resource in the context of a resource group...